Add section about syntax special characters

Hunterness opened the pull request:
We're using a custom syntax for describing the administration commands but we never explain what our special characters mean, for example `{ ... }` or `[ ... ]`.

Hunterness committed:
Update relationship constraint syntax

So that the `[` in the pattern is not confused with the `[` indicating optional

Hunterness committed:
Add section about syntax special characters

Hunterness commented on cypher/cypher-docs/src/docs/dev/ql/administration/security/index.asciidoc:
```diff
@@ -84,6 +85,68 @@ Instead they are defined in terms of their underlying _privileges_ and they can
+Some things to notice about this command is that it includes `{` and `}` in the syntax, and between them has a grouping of either a list of properties or the character `*`.
+It also has multiple optional parts, including the entity part of the command which is the grouping following the graph name.
+
+In difference, there is no need to escape any characters in the index creation command.
+This since `(` and `)` are not special characters and `[` and `]` are not part of the command, but indicate that the index name is optional.
+
+[source, cypher]
+----
+CREATE INDEX [index_name] [IF NOT EXISTS]
+FOR (n:LabelName)
+ON (n.propertyName)
+----
```
Change to node property existence constraint to avoid `OPTIONS` clause needing escaping in 4.2

davidoliverSP2 commented:
Hi @Lojjs - I see you joined and left this PR earlier today - are you able to give a technical review?

davidoliverSP2 commented:
ah, I just saw you've reviewed and left comments on #758

Lojjs reviewed (commented):
@davidoliverSP2 Yes, they were very similar so I just reviewed one. From a technical perspective, the content is correct, but not sure if there is a more pedagogical way to explain it

Hunterness commented on cypher/cypher-docs/src/docs/dev/ql/administration/security/index.asciidoc:
```diff
@@ -84,6 +85,69 @@ Instead they are defined in terms of their underlying _privileges_ and they can
+
+| `\|`
+| Or, used to indicate alternative parts of a command. Needs to be part of a grouping.
+| If the syntax needs to specify either a name or `*`, this can be indicated with `name \| *`.
```
```suggestion
| If the syntax needs to specify either a name or `*`, this can be indicated with `* \| name`.
```
To make the order more consistent (have a card to update other places)

Hunterness commented on cypher/cypher-docs/src/docs/dev/ql/administration/security/index.asciidoc:
```diff
@@ -84,6 +85,69 @@ Instead they are defined in terms of their underlying _privileges_ and they can
+
+| `{` and `}`
+| Used to group parts of the command, common together with `\|`.
+| To use the `or` in the syntax summary it needs to be in a group, `{name \| *}`.
```
```suggestion
| To use the `or` in the syntax summary it needs to be in a group, `{* \| name}`.
```

Hunterness commented on cypher/cypher-docs/src/docs/dev/ql/administration/security/index.asciidoc:
```diff
@@ -84,6 +85,69 @@ Instead they are defined in terms of their underlying _privileges_ and they can
+
+| `"`
+| When a special character is part of the syntax itself, we surround it with `"` to indicate this.
+| To include `{` in the syntax use `"{" { name \| * } "}"`, here we get either `{*}` or `{name}`.
```
```suggestion
| To include `{` in the syntax use `"{" { * \| name } "}"`, here we get either `{*}` or `{name}`.
```

Hunterness commented on cypher/cypher-docs/src/docs/dev/ql/administration/security/index.asciidoc:
```diff
@@ -84,6 +85,69 @@ Instead they are defined in terms of their underlying _privileges_ and they can
+----
+GRANT READ
+    "{" { * | property[, ...] } "}"
+    ON GRAPH[S] {name[, ...] | *}
```
```suggestion
    ON GRAPH[S] {* | name[, ...]}
```

Hunterness committed:
Change order of name and *

To make the order more consistent in the Administration chapter (have a card to update other places)

Hunterness commented on cypher/cypher-docs/src/docs/dev/ql/administration/security/index.asciidoc:
```diff
@@ -84,6 +85,69 @@ Instead they are defined in terms of their underlying _privileges_ and they can
+----
+GRANT READ
+    "{" { * | property[, ...] } "}"
+    ON GRAPH[S] {* | name[, ...]}
```
```suggestion
    ON GRAPH[S] { * | name[, ...] }
```

Hunterness committed:
Add spaces for consistency

Hunterness committed:
Clean-up names

to be consistent with https://github.com/neo4j/neo4j-documentation/pull/825

davidoliverSP2 committed:
editorial update

davidoliverSP2 reviewed (approved):
I pushed the editorial updates from #754 and will merge when tests are green
